File: examples-sdk/java/032-basic-failover-ha/Main.java
// 032-basic-failover-ha — Java SDK equivalent
// Compile: vil compile --from java --input 032-basic-failover-ha/Main.java --release

import dev.vil.*;

public class Main {
    public static void main(String[] args) {
        VilServer server = new VilServer("payment-gateway-ha", 8080);
        ServiceProcess primary = new ServiceProcess("primary");
        primary.endpoint("GET", "/health", "primary_health");
        primary.endpoint("POST", "/charge", "primary_charge");
        server.service(primary);
        ServiceProcess backup = new ServiceProcess("backup");
        backup.endpoint("GET", "/health", "backup_health");
        backup.endpoint("POST", "/charge", "backup_charge");
        server.service(backup);
        server.compile();
    }
}

File: examples-sdk/csharp/032-basic-failover-ha/Main.cs
// 032-basic-failover-ha — C# SDK equivalent
// Compile: vil compile --from csharp --input 032-basic-failover-ha/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("payment-gateway-ha", 8080);
var primary = new ServiceProcess("primary");
primary.Endpoint("GET", "/health", "primary_health");
primary.Endpoint("POST", "/charge", "primary_charge");
server.Service(primary);
var backup = new ServiceProcess("backup");
backup.Endpoint("GET", "/health", "backup_health");
backup.Endpoint("POST", "/charge", "backup_charge");
server.Service(backup);
server.Compile();
